File: src/main/java/com/speechpro/empbase/empbase/service/ContactService.java
package com.speechpro.empbase.empbase.service;

import com.speechpro.empbase.empbase.model.entities.Contact;
import com.speechpro.empbase.empbase.model.entities.ContactType;
import com.speechpro.empbase.empbase.model.entities.Employee;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface ContactService {
    List<Contact> getAll();
    List<Contact> getByEmployee(Employee employee);
    Contact getById(Long id);
    Contact create(Contact contact);
    Contact update(Contact contact);
    void delete(Contact contact);
}

File: src/main/angular/empbase/src/app/services/contact.service.ts
import { Injectable } from '@angular/core';
import { Contact } from '../models/contact';
import { ContactType } from '../models/contactType';
import { Http }    from '@angular/http';

@Injectable()
export class ContactService{

    constructor(private http: Http){}

    getContact(id:number){
        return this.http.get('/api/contact/' + id);
    }

    create(contact:Contact, empId:number){
        if(empId) return this.http.post('/api/contact/' + empId, contact);
    }

    update(contact:Contact){
        return this.http.put('/api/contact', contact);
    }

    delete(id:number){
        if(id) return this.http.delete('/api/contact/' + id);
    }

    getByEmployee(employeeId:number){
        return this.http.get('/api/employee/'+employeeId+'/contact');
    }

    getAllContactTypes(){
        return this.http.get('/api/contact_type/');
    }

    getContactType(id:number){
        return this.http.get('/api/contact_type/' + id);
    }

    createContactType(contactType:ContactType){
        return this.http.post('/api/contact_type', contactType);
    }

    updateContactType(contactType:ContactType){
        return this.http.put('/api/contact_type', contactType);
    }

    deleteContactType(id:number){
        if(id) return this.http.delete('/api/contact_type/' + id);
    }

}